### ISSUE TITLE
midiratio could be more precise

### ISSUE THREAD

**Issue by elgiano** [supercollider/supercollider#5016]
Not a big issue but sometimes annoying, especially in obvious cases:
```supercollider
12.midiratio // -> 1.9999999999945
2.pow(12/12) // -> 2.0 
```
When I think about Henry Partch (roughly) saying that equal temperament is a lie compared to just intonation, this gets particularly annoying :)

Looking at the source code, it seems like midicps/cpsmidi use more precise constants than midiratio/cpsoct... anyone know if this is a bug or intended for some reason I can't understand?

https://github.com/supercollider/supercollider/blob/f806ace7bd8565dd174e7d47a1b32aaa4175a46e/include/plugin_interface/SC_InlineUnaryOp.h#L283

https://github.com/supercollider/supercollider/blob/f806ace7bd8565dd174e7d47a1b32aaa4175a46e/include/plugin_interface/SC_InlineUnaryOp.h#L290
## Environment

* SuperCollider version: 3.11.1
* Operating system: Arch Linux

## Steps to reproduce

```supercollider
12.midiratio // -> 1.9999999999945
2.pow(12/12) // -> 2.0 
```

## Expected vs. actual behavior
I would expect 12.midiratio to be exactly 2.0

**Comment by brianlheim** [comment]
this looks like a bug to me! possibly just missed in https://github.com/supercollider/supercollider/commit/4249e321. that other code is from the initial commit.

however, this stuff is in the public plugin interface (`include/plugin_interface`), which means we need to be a bit careful about how to fix things there...

**Comment by simonperrier11** [comment]
Just came across the same issue in 3.13.0, macOS 13.5.2

**Comment by Spacechild1** [comment]
Yes, `sc_midiratio` should use a more precise constant. This wouldn't break the plugin interface btw.
